content/src/main/content/jcr_root/apps/acs-commons/components/utilities/report-builder/parameters/dynamic-select/datasource/datasource.jsp

```diff
@@ -0,0 +1,86 @@
+<%@page session="false" import="
```

Comment:
hi @DXSaurabh , cany chance you could port this over to a SlingSafeMethodsServlet?

We have a handy DataSourceBuilder API [1] in ACS Commons that can help w some of this boilerplate code. 

Recognizing the irony that im using a JSP as a reference for how to use this API :) ..you can check out [2]

It almost might be cool to add another field that lets a user specify a Query languages and you use ACS's QueryHelper [3] to construct and execute the query in any of the supported languages -- though it looks like youre parsing queries by hand? ..  



[1] https://github.com/Adobe-Consulting-Services/acs-aem-commons/blob/master/bundle/src/main/java/com/adobe/acs/commons/wcm/datasources/DataSourceBuilder.java
[2] https://github.com/Adobe-Consulting-Services/acs-aem-commons/blob/master/content/src/main/content/jcr_root/apps/acs-commons/datasources/named-image-transforms/named-image-transforms.jsp#L27
[3] https://github.com/Adobe-Consulting-Services/acs-aem-commons/blob/master/bundle/src/main/java/com/adobe/acs/commons/util/QueryHelper.java

Reply:
Hi @davidjgonzalez , 

Thanks for reviewing the code.
I have ported the dataSource.jsp over to a SlingSafeMethodsServlet. Adding the one more field for query type is really a good idea but i can add this thing in commit of the new feature.

Thanks